django/thunderstore/repository/comment.py

```diff
@@ -0,0 +1,80 @@
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from thunderstore.community.models import PackageListing
+from thunderstore.repository.models import Comment
+
+
+def clean_content(content: str) -> str:
+    return content.strip()
+
+
+class CreateCommentForm(forms.ModelForm):
+    class Meta:
+        model = Comment
+        fields = ["content"]
+
+    def __init__(
+        self,
+        user: User,
+        commented_object: PackageListing,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.commented_object = commented_object
+
+    def clean_content(self) -> str:
+        return clean_content(self.cleaned_data["content"])
+
+    def save(self, *args, **kwargs) -> Comment:
+        self.instance.author = self.user
+        self.instance.commented_object = self.commented_object
+        return super().save(*args, **kwargs)
+
+
+class EditCommentForm(forms.Form):
+    content = forms.CharField(max_length=Comment._meta.get_field("content").max_length)
+    is_pinned = forms.BooleanField(required=False)
+
+    def __init__(self, user: User, comment: Comment, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.comment = comment
+
+    def can_edit_content(self) -> bool:
+        # Only the comment author can edit a comment's content
+        return self.user == self.comment.author
+
+    def can_pin(self) -> bool:
+        commented_object = self.comment.commented_object
+        if isinstance(commented_object, PackageListing):
+            # Must be a member of the identity to pin
+            return commented_object.package.owner.members.filter(
+                user=self.user,
+            ).exists()
+        raise NotImplementedError()
+
+    def clean_content(self) -> str:
```

Comment:
Same question as the previous clean question; does this prevent superclass default validation on the field?

Reply:
See https://github.com/risk-of-thunder/Thunderstore/pull/196#discussion_r558390658